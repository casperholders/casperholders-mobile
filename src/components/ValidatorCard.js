import CardWithIcons from '@/components/common/CardWithIcons';
import ValidatorIcon from '@/components/common/ValidatorIcon';
import formatCasperAmount from '@/helpers/formatCasperAmount';
import { StyleSheet, View } from 'react-native';
import { Caption, Divider, Text, useTheme } from 'react-native-paper';

/**
 * Display validator infos
 * @param validator
 * @param onPress
 * @returns {JSX.Element}
 * @constructor
 */
export default function ValidatorCard({ validator, onPress }) {
  const theme = useTheme();
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: 200,
    },
    contentContainer: {
      backgroundColor: '#0d0e35',
    },
    itemContainer: {
      padding: 6,
      margin: 6,
      backgroundColor: '#00126b',
    },
    validatorName: {
      paddingHorizontal: 8,
      paddingVertical: 4,
      borderRadius: 8,
      alignSelf: 'flex-start',
    },
    validatorDetails: {
      marginLeft: 4,
    },
  });
  return (
    <>
      <CardWithIcons
        left={validator.png ? <ValidatorIcon
          url={validator.png}
          size={35}
          left
        /> : undefined}
        onPress={onPress}
      >
        <View style={{ ...styles.validatorName, backgroundColor: theme.colors.background }}>
          <Text>
            {validator.name}
          </Text>
        </View>
        {
          validator.name !== validator.publicKey &&
          <Caption style={styles.validatorDetails}>
            {validator.publicKey}
          </Caption>
        }
        <Caption style={styles.validatorDetails}>
          {validator.group} - {validator.delegation_rate}% Fee
          - {formatCasperAmount(validator.staked_amount)} CSPR Staked
        </Caption>
      </CardWithIcons>
      <Divider />
    </>
  );
}
