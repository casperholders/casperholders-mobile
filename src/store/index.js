import authReducer from '@/store/reducers/authReducer';
import networkReducer from '@/store/reducers/networkReducer';
import operationsReducer from '@/store/reducers/operationsReducer';
import { configureStore } from '@reduxjs/toolkit';

export default configureStore({
  reducer: {
    auth: authReducer,
    operations: operationsReducer,
    network: networkReducer,
  },
});
