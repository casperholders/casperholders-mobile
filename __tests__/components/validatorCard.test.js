import ValidatorCard from '@/components/ValidatorCard';
import { resetStore } from '@/store';
import { render } from '@testing-library/react-native';

beforeEach(() => {
  resetStore();
});

jest.setTimeout(500000);

describe('Validator card', () => {
  test('should render Validator card', () => {
    const validator = {
      png: 'https://node.casperholders.com/.well-known/casper/squarelogo256.png',
      name: 'CasperHolders',
      publicKey: '0124bfdae2ed128fa5e4057bc398e4933329570e47240e57fc92f5611a6178eba5',
      group: 'Active',
      delegation_rate: '0',
      staked_amount: '253.00',
    };

    const { queryByText, toJSON } = render(<ValidatorCard validator={validator} />);
    expect(queryByText(/CasperHolders/i)).toBeTruthy();
    expect(queryByText(/0124bfdae2ed128fa5e4057bc398e4933329570e47240e57fc92f5611a6178eba5/i)).toBeTruthy();
    expect(queryByText(/Active/i)).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });
});