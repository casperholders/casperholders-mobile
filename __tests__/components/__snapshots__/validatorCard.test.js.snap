// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator card should render Validator card 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 4,
        "elevation": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
      testID="0124bfdae2ed128fa5e4057bc398e4933329570e47240e57fc92f5611a6178eba5"
    >
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 16,
            },
            Object {
              "paddingVertical": 16,
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "paddingLeft": 0,
              "paddingRight": 12,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 17.5,
                  "height": 35,
                  "width": 35,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://node.casperholders.com/.well-known/casper/squarelogo256.png",
                }
              }
              style={
                Object {
                  "borderRadius": 17.5,
                  "height": 35,
                  "width": 35,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#f6f6f6",
                "borderRadius": 8,
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  undefined,
                ]
              }
            >
              CasperHolders
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "letterSpacing": 0.4,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "marginLeft": 4,
                    },
                  ],
                ],
              ]
            }
          >
            0124bfdae2ed128fa5e4057bc398e4933329570e47240e57fc92f5611a6178eba5
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "letterSpacing": 0.4,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "marginLeft": 4,
                    },
                  ],
                ],
              ]
            }
          >
            Active
             - 
            0
            % Fee - 
            - CSPR
             CSPR Staked
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
          "height": 0.5,
        },
        undefined,
        undefined,
      ]
    }
  />,
]
`;
